ocs_ci/ocs/scale_pgsql.py

```diff
@@ -0,0 +1,135 @@
+"""
+ScalePodPGSQL workload class for scale
+"""
+import logging
+
+from tests import helpers
+from ocs_ci.ocs.pgsql import Postgresql
+from ocs_ci.utility import templating
+from ocs_ci.ocs import constants, machine
+from ocs_ci.framework import config
+from ocs_ci.ocs.exceptions import UnavailableResourceException
+
+log = logging.getLogger(__name__)
+
+
+class ScalePodPGSQL(Postgresql):
+    """
+    Scale Postgresql workload with scale parameters and functions
+    """
+
+    def __init__(self, node_selector=constants.SCALE_NODE_SELECTOR, **kwargs):
+        """
+        Initializer function
+
+        """
+        super().__init__(**kwargs)
+        self._node_selector = node_selector
+        self.ms_name = list()
+
+    def apply_crd(self, crd):
+        Postgresql.apply_crd(self, crd=crd)
+
+    def setup_postgresql(self, replicas, node_selector=None):
+        # Node selector for postgresql
+        pgsql_sset = templating.load_yaml(
+            constants.PGSQL_STATEFULSET_YAML
+        )
+        if node_selector is not None:
+            pgsql_sset['spec']['template']['spec'][
+                'nodeSelector'] = node_selector
+        Postgresql.setup_postgresql(self, replicas=replicas)
+
+    def _create_pgbench_benchmark(
+        self, replicas, clients=None, threads=None,
+        transactions=None, scaling_factor=None,
+        timeout=None
+    ):
+        Postgresql.create_pgbench_benchmark(
+            self, replicas, clients=clients,
+            threads=threads, transactions=transactions,
+            scaling_factor=scaling_factor, timeout=timeout
+        )
+
+    def _wait_for_pgbench_status(self, status, timeout=None):
+        Postgresql.wait_for_postgres_status(
+            self, status=status, timeout=timeout
+        )
+
+    def _get_pgbench_pods(self):
+        Postgresql.get_pgbench_pods()
+
+    def _validate_pgbench_run(self, pgbench_pods, print_table=True):
+        Postgresql.validate_pgbench_run(self, pgbench_pods, print_table=True)
+
+    def add_worker_node(self, instance_type=None):
```

Comment:
wouldn't it be helpful if this function is outside the pgsql class?

Reply:
@tiffanyn108 we can sync-up on how to re-use/enhace the code from function check_and_add_enough_worker(worker_count) https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/scale_lib.py#L541

Reply:
Moved add_worker_node() and delete_worker_node() out of pgsql class.